apps/client/src/module/auth/register-form.tsx

```diff
@@ -12,16 +12,51 @@ import { AuthFormContainer } from "./auth-form-container";
 import { register, RegisterPayload } from "./auth-service";
 import { AuthRoute } from "./route";
 import { UserHelpMsgs } from "@src/utils/user-help-messgage";
+import {
+  displayNamePatten,
+  emailPattern,
+  passwordPatten,
+  DisplayErrorMsgs,
+  EmailErrorMsgs,
+  PasswordErrorMsgs,
+  ConfirmPasswordErrorMsgs,
+  RefErrorMsgs,
+} from "@src/utils/validation-message";
 
 type FormSchema = z.infer<typeof formSchema>;
 const formSchema = z.object({
-  displayName: z.string(),
-  email: z.string().email(),
+  displayName: z.union([
+    z.string()
+      .min(4, { message: DisplayErrorMsgs.MinLength })
+      .max(15, { message: DisplayErrorMsgs.MaxLength })
+      .regex(displayNamePatten, {
+        message: DisplayErrorMsgs.Invalid,
+      }),
+    z.string()
+      .length(0)
+      .transform(() => "New User"),
+  ]),
+  email: z.string()
+    .min(1, { message: EmailErrorMsgs.Required })
+    .email({ message: EmailErrorMsgs.Invalid })
+    .regex(emailPattern, { message: EmailErrorMsgs.Invalid }),
   mobile: z.string().optional(),
-  password: z.string().min(8),
-  confirmPassword: z.string().min(8),
+  password: z.string()
+    .min(1, { message: PasswordErrorMsgs.Required })
+    .min(8, { message: PasswordErrorMsgs.MinLength })
+    .max(32, { message: PasswordErrorMsgs.MaxLength })
+    .regex(passwordPatten, { message: PasswordErrorMsgs.Invalid }),
+    confirmPassword: z.string()
+    .min(1, { message: ConfirmPasswordErrorMsgs.Required })
+    .min(8, { message: PasswordErrorMsgs.MinLength })
+    .max(32, { message: PasswordErrorMsgs.MaxLength })
+    .regex(passwordPatten, { message: PasswordErrorMsgs.Invalid }),
   realName: z.string().optional(),
-  ref: z.string(),
+  ref: z.string().min(1, { message: RefErrorMsgs.Required }),
```

Comment:
Oh. Do we have a default value for this?

Reply:
yeah we have a pre-filled default value corresponding to the backend, I‘ll add a default value specification here